sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubConsumer.cs

```diff
@@ -237,5 +348,260 @@ protected EventHubConsumer()
         ///
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString() => base.ToString();
+
+        /// <summary>
+        ///   Creates a subscription to events being published to the partition associated with
+        ///   the consumer.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the event hub with the subscription; used for informational purposes only.</param>
+        /// <param name="partitionId">The identifier for the partition to associate with the subscription; used for informational purposes only.</param>
+        /// <param name="consumerGroup">The name of the consumer group to associate with the subscription; used for informational purposes only.</param>
+        /// <param name="maximumQueuedEvents">The maximum number of events to queue while waiting for them to be read.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to signal the request to cancel the event enumeration.</param>
+        ///
+        /// <returns>The elements of a channel subscription, including its identifier and a reader for the subscribed channel.</returns>
+        ///
+        private (Guid Identifier, ChannelReader<EventData> ChannelReader) SubscribeToChannel(string eventHubName,
+                                                                                             string partitionId,
+                                                                                             string consumerGroup,
+                                                                                             int maximumQueuedEvents,
+                                                                                             CancellationToken cancellationToken)
+        {
+            var channel = CreateEventChannel(maximumQueuedEvents);
+            var identifier = Guid.NewGuid();
+
+            if (!ActiveChannels.TryAdd(identifier, channel))
```

Comment:
This would fail only if guids collide or am I missing something?

Reply:
That's the only case that I can discern, unless something internal to the concurrent collection gets angry.

Reply:
`Assert.True` :) ?

Reply:
Interesting thought;  I went this approach because I want to ensure retries.  The chance of this path ever getting hit is very small and the chance of it happening more than once...